butte-build/src/codegen.rs

```diff
@@ -190,14 +166,48 @@ fn to_type_token(
     }
 }
 
+/// A trait to format literals according to a provided type.
+trait PrimitiveLiteralToken {
+    /// The type of the token that will be returned formatted according to the provided type.
+    type TokenType;
+
+    fn fmt_lit(&self, ty: impl Spanned + fmt::Display) -> Self::TokenType;
+}
+
+impl PrimitiveLiteralToken for ast::IntegerConstant {
+    type TokenType = syn::LitInt;
+
+    fn fmt_lit(&self, ty: impl Spanned + fmt::Display) -> Self::TokenType {
+        Self::TokenType::new(&format!("{}{}", self, ty), ty.span())
+    }
+}
+
+impl PrimitiveLiteralToken for ast::FloatingConstant {
+    type TokenType = syn::LitFloat;
+
+    fn fmt_lit(&self, ty: impl Spanned + fmt::Display) -> Self::TokenType {
+        Self::TokenType::new(
+            &match ty.to_string().as_str() {
+                float_ty @ "f32" => format!("{}{}", *self as f32, float_ty),
+                float_ty => format!("{}{}", self, float_ty),
+            },
+            ty.span(),
+        )
+    }
+}
+
 /// Convert a `types::ast::DefaultValue` to a default field value
 fn to_default_value(
     arg_ty: &impl ToTokens,
     default_value: &ast::DefaultValue<'_>,
-) -> impl ToTokens {
+) -> impl ToTokens + fmt::Debug + ToString {
     match default_value {
         // Scalar field
-        ast::DefaultValue::Scalar(s) => s.to_token_stream(),
+        ast::DefaultValue::Scalar(s) => match s {
+            ast::Scalar::Integer(i) => i.fmt_lit(arg_ty.to_token_stream()).to_token_stream(),
+            ast::Scalar::Float(f) => f.fmt_lit(arg_ty.to_token_stream()).to_token_stream(),
+            ast::Scalar::Boolean(b) => b.to_token_stream(),
```

Comment:
Why is this pattern different only for `Boolean`?

Reply:
Because there's no need to do any special formatting for `Boolean` values, the syntax of Rust boolean literals is identical to the syntax of Flatbuffer boolean literals.